pkg/cmd/roachtest/cluster.go

```diff
@@ -2303,6 +2303,20 @@ func (c *clusterImpl) ConnE(ctx context.Context, node int) (*gosql.DB, error) {
 	return db, nil
 }
 
+// ConnEAsUser returns a SQL connection to the specified node as a specific user
+func (c *clusterImpl) ConnEAsUser(ctx context.Context, node int, user string) (*gosql.DB, error) {
+	urls, err := c.ExternalPGUrl(ctx, c.Node(node))
+	if err != nil {
+		return nil, err
+	}
+	dataSourceName := strings.Replace(urls[0], "root", user, 1)
```

Comment:
Please parse the url (`url.Parse`) and replace the username properly (`u.User`). Then `u.String()` is the result.

Reply:
Thanks for catching this! I resolved it in my other PR https://github.com/cockroachdb/cockroach/pull/72982